src/sqlstore/timeline.py

```diff
@@ -0,0 +1,234 @@
+
+    def __init__(self, platformId: str, gameId: int, frameInterval: int):
+        self.platformId = platformId
+        self.gameId = gameId
+        self.frameInterval = frameInterval
+
+    def __repr__(self):
+        return f"{self.platformId}_{self.gameId} with interval {self.frameInterval}ms"
+
+
+class SQLTimelineFrame(Base):
+    __tablename__ = "match_timeline_frame"
+
+    platformId = Column(String(7), primary_key=True)
+    gameId = Column(BigInteger, primary_key=True)
+    frameId = Column(Integer, primary_key=True)     # this is not present in the data and needs to be calculated
+    # eventIds = Column(Integer, ForeignKey("match_timeline_event.eventId"))  # TODO: is this relevant here?
+    timestamp = Column(Integer, nullable=False)
+    timeCreated = Column("timeCreated", DateTime(timezone=True), server_default=func.now())
+    lastUpdate = Column("lastUpdate", DateTime(timezone=True), onupdate=func.now())
+
+    def __init__(self, platformId: str, gameId: int, frameId: int, timestamp: int):
+        self.platformId = platformId
+        self.gameId = gameId
+        self.frameId = frameId
+        self.timestamp = timestamp
+
+    def __repr__(self):
+        return f"{self.platformId}_{self.gameId} frame {self.frameId} at {self.timestamp}"
+
+
+class SQLTimelineEvent(Base):
+    __tablename__ = "match_timeline_event"
+
+    platformId = Column(String(7), primary_key=True)
+    gameId = Column(BigInteger, primary_key=True)
+    frameId = Column(Integer, primary_key=True)
+    eventId = Column(Integer, primary_key=True)
+    timestamp = Column(Integer)
+    type = Column(String(100))
+    participantId = Column(Integer)
+    itemId = Column(Integer)
+    skillSlot = Column(Integer)
+    creatorId = Column(Integer)
+    teamId = Column(Integer)
+    afterId = Column(Integer)
+    beforeId = Column(Integer)
+    wardType = Column(String(50))
+    timeCreated = Column("timeCreated", DateTime(timezone=True), server_default=func.now())
+    lastUpdate = Column("lastUpdate", DateTime(timezone=True), onupdate=func.now())
+
+    def __init__(self, **kwargs):
+        for attr in ('platformId', 'gameId', 'frameId', 'eventId', 'timestamp', 'type', 'participantId', 'itemId',
+                     'skillSlot', 'creatorId', 'teamId', 'afterId', 'beforeId', 'wardType'):
+            setattr(self, attr, kwargs.get(attr))
+
+    def __repr__(self):
+        return f"{self.platformId}_{self.gameId} frame {self.frameId} event {self.eventId} at {self.timestamp} of type {self.type}"
+
+
+class SQLTimelineKillEvent(Base):
+    __tablename__ = "match_timeline_event_kill"
+    platformId = Column(String(7), primary_key=True)
+    gameId = Column(BigInteger, primary_key=True)
+    frameId = Column(Integer, primary_key=True)
+    killId = Column(Integer, Identity(always=True), primary_key=True)
+    # eventId = Column(Integer, ForeignKey("match_timeline_event.eventId"))   # TODO: is this correct?
```

Comment:
I think so, yes. one-to-one rtelationship between kill event and general event. I guess that is true.